src/components/edit-form/edit-form.tsx

```diff
@@ -1,49 +1,170 @@
+import { useContext } from 'react';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { useDispatch, useSelector } from 'react-redux';
+import { ThunkDispatch } from 'redux-thunk';
+import { AnyAction } from 'redux';
 import Button from '../button/button';
 import { genres } from '../main/info';
-
+import AppContext from '../app-context/app-context';
+import { FormInput } from '../../types/form-input';
 import './style.css';
+import postMovie from '../../state/post-movie';
+import RootState from '../../types/root-state';
+import putMovie from '../../state/put-movie';
+
+type EditFormProps = {
+  onModalClose: () => void;
+};
+
+export default function EditForm({ onModalClose }: EditFormProps) {
```

Comment:
I think it is worth creating two separate components for Add and Edit.

Reply:
Created two components AddForm and EditForm.